### PR TITLE
Update syn to 0.11

### PR DESCRIPTION
**Pull request by nox**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -12,5 +12,5 @@ keywords = ["syn", "macros", "derive", "expand_substructure", "enum"]
 
 
 [dependencies]
-syn = "0.10"
+syn = "0.11"
```
Comment (dtolnay) [inline Cargo.toml]: This could be ">=0.10, <0.12" to keep synstructure at 0.4.

**Comment by mystor** [comment]
I merged this in 35db51e - /me is bad at github stuff so apparently I didn't successfully make it close this issue automatically.

File: Cargo.toml (hunk 1 above)
Comment (nox) [inline Cargo.toml]: This kind of thing breaks Cargo and it's lets-use-the-latest-version-ever-all-the-time-even-if-it-introduces-multiple-versions-of-syn behaviour.

**Comment by nox** [comment]
This should have been a breaking change.

**Comment by nox** [comment]
If my project depends on syn 0.10 and synstructure 0.4 and I update synstructure, the build will fail because synstructure 0.4.1 will use syn 0.11 and my project will pass to it syn 0.10 types.

**Comment by nox** [comment]
@mystor We just wasted time because of this, think we could yank 0.4.1 and publish 0.5 with syn 0.11 only?

**Comment by mystor** [comment]
@nox I am currently at a cottage without reliable internet. I can do that next time I have reliable enough internet.

**Comment by nox** [comment]
Sure! Have a pleasant time at the cottage. :)

**Comment by mystor** [comment]
@nox 0.4.1 has been yanked, and 0.5 has been published :)

**Comment by nox** [comment]
Thanks @mystor, well appreciated.

**Pull request closed without merging**